glusterfs/tests/conftest.py

```diff
@@ -48,29 +47,37 @@ def config():
 
 @pytest.fixture()
 def mock_gstatus_data():
-    f_name = os.path.join(os.path.dirname(__file__), 'fixtures', 'gstatus.json')
+    f_name = os.path.join(os.path.dirname(__file__), 'fixtures', 'gstatus.txt')
     with open(f_name) as f:
         data = f.read()
-    with mock.patch('datadog_checks.glusterfs.check.get_subprocess_output', return_value=(data, '', 0)):
+
+    with mock.patch('datadog_checks.glusterfs.check.GlusterfsCheck._get_gstatus_output', return_value=(data, "", 0)):
         yield
 
 
 def create_volume():
-    run_command("docker exec gluster-node-2 mkdir -p /export-test", capture=True, check=True)
-
-    for command in (
-        'gluster peer probe gluster-node-2',
-        'mkdir -p /export-test',
-        'gluster volume create gv0 replica 2 gluster-node-1:/export-test gluster-node-2:/export-test force',
-        'gluster volume start gv0',
-        'yum update -y',
-        'yum install -y python3',
-        'curl -LO https://github.com/gluster/gstatus/releases/download/v1.0.5/gstatus',
-        'chmod +x ./gstatus',
-        'mv ./gstatus /usr/local/bin/gstatus',
-    ):
-        run_command("docker exec gluster-node-1 {}".format(command), capture=True, check=True)
-        time.sleep(10)
+    # The image for some reason doesn't actually start glusterd. So the below two lines will start it manually.
+    run_command(
+        "docker exec gluster-node-2 /usr/sbin/glusterd -p /var/run/glusterd.pid --log-level INFO",
+        capture=True,
+        check=False,
+    )
+    run_command(
+        "docker exec gluster-node-1 /usr/sbin/glusterd -p /var/run/glusterd.pid --log-level INFO",
+        capture=True,
+        check=False,
+    )
```

Comment:
We can also put this into the `Dockerfile`, have a look at https://docs.docker.com/reference/dockerfile/#entrypoint OR https://docs.docker.com/reference/dockerfile/#cmd

Reply:
I tried this actually. But it ended up causing the containers to just not spin up. I'm not sure why. I think with entrypoint and cmd, it causes the container to ignore the default startup commands and only run the command you specify so I wasn't able to get this to work, because I would ideally want it to run what the container image would normally do and then run these to start the daemon manually.

Reply:
Oh I get it. Mind adding this as an inline comment? Can be short: "Tried these commands as entrypoint/cmd, but the container image had some other entrypoint/cmd that could not be overridden without crashing the container".